plugins/inputs/logfile/tailersrc_test.go

```diff
@@ -179,10 +180,11 @@ func TestOffsetDoneCallBack(t *testing.T) {
 			IsUTF16:     false,
 		})
 
-	if err != nil {
-		t.Errorf("Failed to create tailer src for file %v with error: %v", file, err)
-		return
-	}
+	assert.NoError(t, err)
+
+	//Increase number of monitored files by 1
+	numUsedFds.Acquire()
```

Comment:
Do we need to test this fd counter in every test? 

If we do, then we need to remove this code duplication. Either we should pull out the entire part where we create the tailer, `tailer, err := tail.TailFile()` into a separate test function, where we instantiate the semaphore and increment it by one there, or just have a separate function that creates the semaphore where it has already acquired once and call that. There's not much reason to copy paste this `Acquire` call and accompanying comment into every test

Reply:
FYi, there is a function for [setup tailer](https://github.com/aws/amazon-cloudwatch-agent/blob/master/plugins/inputs/logfile/tail/tail_test.go#L121-L153). The different between these two are for setting the `ts.SetOutput(func(evt logs.LogEvent)`